cloudwatch_metrics_common/include/cloudwatch_metrics_common/utils/metric_object.h

```diff
@@ -46,17 +47,53 @@ static std::unordered_map<std::string, Aws::CloudWatch::Model ::StandardUnit> un
   {"", Aws::CloudWatch::Model::StandardUnit::None},
 };
 
+enum class StatisticValuesType
+{
+  MINIMUM,
+  MAXIMUM,
+  SUM,
+  SAMPLE_COUNT
+};
+
 /**
  * Wrapper object for the AWS specific Aws::CloudWatch::Model::MetricDatum. This object is meant to be constructed from
  * userland provided metric data instead of using the AWS SKD specific object.
  */
 struct MetricObject {
-    const std::string metric_name;
-    const double value;
-    const std::string unit;
-    const int64_t timestamp;
-    const std::map<std::string, std::string> dimensions;
-    const int storage_resolution;
+  MetricObject()
```

Comment:
Why is this empty constuctor needed? Shouldn't metric data be defined upon construction?

Reply:
Fine, I'll remove this convenience function.